### PR TITLE
MAISTRA-2557 add support for processing v1 CRDs

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

File: pkg/bootstrap/crds.go [hunk 1]
```diff
@@ -181,35 +182,52 @@ func processCRDFile(ctx context.Context, cl client.Client, fileName string) erro
...
 	}
-	obj := &apiextensionsv1beta1.CustomResourceDefinition{}
-	_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawJSON, nil, obj)
+	v1beta1obj := &apiextensionsv1beta1.CustomResourceDefinition{}
+	_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawJSON, nil, v1beta1obj)
 	if err != nil {
-		log.Error(err, "unable to decode object into Unstructured")
-		return nil, err
+		v1obj := &apiextensionsv1.CustomResourceDefinition{}
+		_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawJSON, nil, v1beta1obj)
```
Comment (jwendell) [inline pkg/bootstrap/crds.go]: should be `v1obj`, no?

File: pkg/bootstrap/crds.go [hunk 2]
```diff
@@ -181,35 +182,52 @@ func processCRDFile(ctx context.Context, cl client.Client, fileName string) erro
...
+		if v1obj.GroupVersionKind().GroupKind().String() == "CustomResourceDefinition.apiextensions.k8s.io" {
+			return v1obj, nil
+		}
+		return nil, nil
 	}
-	if obj.GroupVersionKind().GroupKind().String() == "CustomResourceDefinition.apiextensions.k8s.io" {
-		return obj, nil
-	} else {
+	if v1beta1obj.GroupVersionKind().GroupKind().String() != "CustomResourceDefinition.apiextensions.k8s.io" {
 		return nil, nil
```
Comment (jwendell) [inline pkg/bootstrap/crds.go:210]: should return an error?

File: pkg/controller/common/test/api.go [hunk 3]
```diff
@@ -148,26 +150,33 @@ func (a *AbstractActionFilter) Handles(action clienttesting.Action) bool {
...
-    return a
+	resourceAndSub := strings.SplitN(resource, "/", 2)
+	a.Resource = resourceAndSub[0]
+	if len(resourceAndSub) > 1 {
+		a.Subresource = resourceAndSub[1]
+	}
+	return a
+}
+
+// In initializes the version whithin which the filter should apply.  Use "*"
```
Comment (jwendell) [inline pkg/controller/common/test/api.go]: s/In/Version/

File: pkg/controller/common/test/assertions.go [hunk 4]
```diff
@@ -34,22 +34,29 @@ type ActionAssertionFactory struct {
 // between resource an subresource, e.g. deployments/status.  Use "*" to match
 // all resources.
 func (f *ActionAssertionFactory) On(resource string) *ActionAssertionFactory {
-    f.AbstractActionFilter.On(resource)
-    return f
+	f.AbstractActionFilter.On(resource)
+	return f
+}
+
+// In initializes the version whithin which the created assertion should apply.
```
Comment (jwendell) [inline pkg/controller/common/test/assertions.go]: s/In/Version/

File: pkg/controller/common/test/verifiers.go [hunk 5]
```diff
@@ -40,6 +40,13 @@ func (f *ActionVerifierFactory) On(resource string) *ActionVerifierFactory {
 	return f
 }
 
+// In initializes the version whithin which the created verifier should apply.
```
Comment (jwendell) [inline pkg/controller/common/test/verifiers.go]: s/In/Version/

File: pkg/bootstrap/crds.go (hunk 2 above)
Comment (rcernich) [reply pkg/bootstrap/crds.go:210]: That would be changing the original behavior.  If we return an error, then CRD processing fails as will all reconciles from this point on.  I believe all this does is silently fails to keep reconcile going, without breaking everything, as the failure is unrecoverable (i.e. no user action can fix it).  I'll add a log message though, so there's something visible.

File: pkg/bootstrap/crds.go (hunk 1 above)
Comment (rcernich) [reply pkg/bootstrap/crds.go]: good catch!

**Review by dgn — APPROVED** [review]

**Review by jwendell — APPROVED** [review]

**Comment by rcernich** [comment]
I've been able to verify that existing installs for v1.1, v2.0, and v2.1 all succeed.

**Pull request merged**